src/screens/ChannelListScreen/index.tsx

```diff
@@ -46,17 +44,7 @@
 
   const renderChannelItem = ({item}) => {
     if (item?.channelType === 'PM') {
-      const isFromAnonymous = item?.rawJson?.channel?.channel_type === 4;
-      const hasFollowButton = !isFromAnonymous && !isInitialFollowing(item);
-
-      return (
-        <MessageChannelItem
-          item={item}
-          onChannelPressed={() => goToChatScreen(item)}
-          hasFollowButton={hasFollowButton}
-          handleFollow={() => handleFollow(item)}
-        />
-      );
+      return <MessageChannelItem item={item} onChannelPressed={() => goToChatScreen(item)} />;
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new implementation simplifies the rendering of `MessageChannelItem` by removing checks for anonymous messages and initial following state. This change appears to be consistent with the PR's goal of simplifying follow button logic. However, please ensure that the removed checks are handled elsewhere or are no longer necessary. If these checks are still required, consider reintroducing them in a way that aligns with your new approach.


<!-- This is an auto-generated comment by OSS CodeRabbit -->